okhttp/src/main/java/okhttp3/ConnectionPool.kt

```diff
@@ -29,12 +30,19 @@ import java.util.concurrent.TimeUnit
  * Currently this pool holds up to 5 idle connections which will be evicted after 5 minutes of
  * inactivity.
  */
-class ConnectionPool(
-  maxIdleConnections: Int,
-  keepAliveDuration: Long,
-  timeUnit: TimeUnit
+class ConnectionPool internal constructor(
+  internal val delegate: RealConnectionPool
 ) {
-  internal val delegate = RealConnectionPool(maxIdleConnections, keepAliveDuration, timeUnit)
+  constructor(
+    maxIdleConnections: Int,
+    keepAliveDuration: Long,
+    timeUnit: TimeUnit
+  ) : this(RealConnectionPool(
+      taskRunner = TaskRunner.INSTANCE,
```

Comment:
How will we detect contention of coordinatorExecutor since its now a process singleton thread?

Reply:
I don’t anticipate contention on it because it doesn’t do anything other than kickoff tasks and sleep.

Reply:
... and it should not do anything else unless we address that.

That said, it’d be interesting to shrink the total number of threads for a RealConnectionPool use case from 2 to 1 by sharing threads between task execution and the coordinator. The implementation gets a little more complex because threads would need to switch roles.